cmd/release-notes/main.go

```diff
@@ -149,8 +149,29 @@ func init() {
 	cmd.PersistentFlags().StringVar(
 		&opts.Format,
 		"format",
-		util.EnvDefault("FORMAT", "markdown"),
-		"The format for notes output (options: markdown, json)",
+		util.EnvDefault("FORMAT", notes.FormatOutputMarkdown),
+		fmt.Sprintf("The format for notes output (options: %s)",
+			strings.Join([]string{
+				notes.FormatOutputJSON,
+				notes.FormatOutputMarkdown,
+			}, ", "),
+		),
+	)
+
+	// template is the file path to a markdown template to render notes in.
+	cmd.PersistentFlags().StringVar(
+		&opts.Template,
+		"template",
+		util.EnvDefault("TEMPLATE", ""),
+		fmt.Sprintf("A Go template string to use when rendering notes."),
+	)
+
+	// template is the file path to a markdown template to render notes in.
+	cmd.PersistentFlags().StringVar(
+		&opts.TemplatePath,
+		"template-path",
+		util.EnvDefault("TEMPLATE_PATH", ""),
+		fmt.Sprintf("The absolute path to a file containing a Go template used when rendering notes."),
 	)
```

Comment:
This is probably a result of me not being able to convey what I meant in one of my former comments. Sorry about that, let me try again:

Having both `template` & `template-path` here is probably not really what we want, and I don't think we need that. I think we should use `template`, interpret it as a file path (both relative and absolute, [`filepath.Abs`](https://golang.org/pkg/path/filepath/#Abs) is probably helpful).

----


I am not sure if we even want both `format` & `template`. I am thinking if we want to move to a world where we only have one flag, I'll call that flag `format` for now, which does the following

- `--format=json`: select the built in template "json" and render with that, it happens to render the output as valid JSON
- `--format=md`: select the built in templage "md" and render with that, it happens to render the output as valid Markdown
- `--format=html`: select the built in templage "html" and render with that, it happens to render the output as valid HTML
- `--format=go-template:some/path/to/file.tmpl`: read the external `some/path/to/file.tmpl`, use it as a go-template and render with that, it might render the output in whatever format
- *no `--format` specified*: render with default "md" template, same as "--format=md"

I am not sure if this is a good approach, just throwing it out there ...

Reply:
I don't feel strongly here. I can implement whatever ends up being the consensus.

I do like this idea of stuffing everyone into --format as you suggested, however.

Unless someone says otherwise, I'll implement it as to suggested @hoegaarden and I'll make krel/changelog do the same.